### DISCUSSION TITLE
How does fastapi establish Http long connection?

### DISCUSSION THREAD

**Discussion by zhao004** [fastapi/fastapi#11604] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def root():
    return {"message": "Hello World"}

# How to write Http long connection API?
```

### Description

![图片1](https://github.com/tiangolo/fastapi/assets/82579833/f52c4060-c347-4639-a4df-e2ab8f4fe7ae)

![图片2](https://github.com/tiangolo/fastapi/assets/82579833/46414596-f8b2-4f31-abba-516a6f84e901)
 ### __How to usefastapi to build a long HTTP connection and return data bit by bit like in the picture?__

### Operating System

Windows

### Operating System Details

windows10

### FastAPI Version

fastapi  0.111.0

### Pydantic Version

pydantic   2.7.1

### Python Version

Python 3.11.1

**Comment by YuriiMotov** [comment, marked as answer]
Hi!

You can do it using `StreamingResponse`:
```
from fastapi import FastAPI
from fastapi.responses import StreamingResponse

app = FastAPI()


async def streamer():
    for i in range(10):
        yield b"some data"


@app.get("/")
async def main():
    return StreamingResponse(streamer())
```

In the example above you need to change `streamer` function to wait for the portion of data to send, send it (`yield data_as_bytes`) and wait for the next portion of data.

**Reply by zhao004** [reply]
Thank you very much, Mr. YuriiMotov
